src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.add(this);
+  }
 }
 
+Animal.alive = [];
+
+Animal.add = (animal) => {
+  Animal.alive.push(animal);
+};
+
+Animal.checkAlive = () => {
+  Animal.alive = Animal.alive.filter(
+    ({ health }) => health > 0
+  );
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!(prey instanceof Herbivore) || prey.hidden) {
+      return;
+    }
+
+    prey.health -= 50;
+
+    Animal.checkAlive();
```

Comment:
Let's do it only when we know that there is an animal that have 0 or less health